Rest/alm-vss-client/src/main/java/com/microsoft/alm/client/DefaultRestClientHandler.java

```diff
@@ -0,0 +1,248 @@
+
+        final String routeTemplate = location.getRouteTemplate();
+        final String actualTemplate = removeUndefinedOptionalParameters(routeTemplate, dictionary);
+        final WebTarget targetTemplate = baseTarget.path(actualTemplate);
+
+        WebTarget target = targetTemplate.resolveTemplates(dictionary);
+
+        if (queryParameters != null) {
+            for (final Entry<String, String> queryParameter : queryParameters.entrySet()) {
+                target = target.queryParam(queryParameter.getKey(), queryParameter.getValue());
+            }
+        }
+
+        return target;
+    }
+
+    private MediaType getMediaTypeWithQualityHeaderValue(final String baseMediaType, final ApiResourceVersion version) {
+        final Map<String, String> parameters = new HashMap<String, String>();
+        parameters.put(API_VERSION_PARAMETER_NAME, version.toString());
+        parameters.put(CHARSET_PARAMETER_NAME, StringUtil.UTF8_CHARSET);
+
+        final String[] mediaTypePart = baseMediaType.split("/", 2); //$NON-NLS-1$
+        final MediaType mediaType = new MediaType(mediaTypePart[0], mediaTypePart[1], parameters);
+
+        return mediaType;
+    }
+
+    private MediaType getMediaTypeWithQualityHeaderValue(final String baseMediaType) {
+        final Map<String, String> parameters = new HashMap<String, String>();
+        parameters.put(CHARSET_PARAMETER_NAME, StringUtil.UTF8_CHARSET);
+
+        final String[] mediaTypePart = baseMediaType.split("/", 2); //$NON-NLS-1$
+        final MediaType mediaType = new MediaType(mediaTypePart[0], mediaTypePart[1], parameters);
+
+        return mediaType;
+    }
+
+    public class JaxRsRequest implements VssRestRequest {
+
+        final private Invocation request;
+
+        public JaxRsRequest(final Invocation request) {
+            this.request = request;
+        }
+
+        @Override
+        public VssRestResponse sendRequest() {
+            Response response;
+
+            try {
+                response = request.submit().get();
+            } catch (final ExecutionException e) {
+                // TODO log exception
+                // TODO process cancellation
+
+                throw new VssServiceException(e.getMessage(), e);
+            } catch (final InterruptedException e) {
+                // TODO log exception
+                // TODO process cancellation
+
+                throw new VssServiceException(e.getMessage(), e);
+            }
+
+            return new JaxRsResponse(response);
+        }
+    }
+
+    public class JaxRsResponse implements VssRestResponse {
+
+        final private Response response;
+
+        public JaxRsResponse(final Response response) {
+            this.response = response;
+        }
+
+        @Override
+        public boolean isJsonResponse() {
+            if (response != null && response.getMediaType() != null) {
+                return response.getMediaType().getType().equalsIgnoreCase("application") //$NON-NLS-1$
```

Comment:
Would it be possible to use StringUtils to compare getMediaType().toString() and MediaType.APPLICATION_JSON?